core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -165,8 +165,13 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   final protected def writeReplace(): Any = {
     if (atDriverSide) {
       if (!isRegistered) {
+        val printName = if (this.name.isDefined) {
+          s" '${this.name.get}' "
+        } else {
+          " "
+        }
         throw new UnsupportedOperationException(
-          "Accumulator must be registered before send to executor")
+          s"Accumulator${printName}must be registered before send to executor")
       }
```

Comment:
My point was, this will end up being an empty string - since it is not registered.
Did you try this in some case where it helped ?

Reply:
You're right, if all is done correctly, we never hit this. I hit this when I added some DSV2 metrics and had a bug in my code. It took a while to find the cause.